internal/kubernetes/utils.go

```diff
@@ -189,10 +189,53 @@ func CreateOrUpdateValidatingWebhookConfiguration(ctx context.Context, c client.
 func mergeMetadata(new *metav1.ObjectMeta, old metav1.ObjectMeta) {
 	new.ResourceVersion = old.ResourceVersion
 
+	new.OwnerReferences = restrictControllers(new.OwnerReferences)
```

Comment:
I took over the functionality in a similar way in [this PR](https://github.com/kyma-project/telemetry-manager/pull/127). Revert it here.